### ISSUE TITLE
Function `str` is not valid in a type expressiontyinvalid-type-form

### ISSUE THREAD

**Issue by goodfr** [astral-sh/ty#2970]
Firstly, thanks for creating this really nice tool.

Something new came up using VSCode `ty` extension, using `str` typing yields error. I tried a roll back to previous version `2026.8.0` and `2026.4.0` with no luck.

error message :

> Function `str` is not valid in a type expressionty[invalid-type-form](https://ty.dev/rules#invalid-type-form)

Example of code :

```py
async def get_smt(request: Request, p_id: str, b_id: str, a_id: str) -> Any:
  ...
```

**Comment by charliermarsh** [comment]
Can you come up with a reproduction in our playground? E.g., https://play.ty.dev/0bb56172-7b00-4455-930f-59d0638921cb

**Comment by AlexWaygood** [comment]
I'm guessing you have a `str` method that is clobbering the builtin `str` symbol? in which case this is https://github.com/astral-sh/ty/issues/1747

**Comment by carljm** [comment]
Strongly suspect this is #1747, so will close as duplicate -- feel free to comment if you don't have a method named `str` and the cause appears to be something else.

**Comment by goodfr** [comment]
Thanks for those answers ! It was indeed a silly typo : `from locale import str` at the top of the module added by an addin (Pylance Analysis option perfoms "auto import")...
